rust/lance-namespace-impls/src/rest_adapter.rs

```diff
@@ -76,24 +77,57 @@ impl RestAdapter {
             .with_state(self.backend.clone())
     }
 
-    /// Start the REST server (blocking)
-    pub async fn serve(self) -> Result<()> {
+    /// Start the REST server in the background and return a handle for shutdown.
+    ///
+    /// This method binds to the configured address and spawns a background task
+    /// to handle requests. The returned handle can be used to gracefully shut down
+    /// the server.
+    ///
+    /// Returns an error immediately if the server fails to bind to the address.
+    pub async fn start(self) -> Result<RestAdapterHandle> {
         let addr = format!("{}:{}", self.config.host, self.config.port);
-        let listener = tokio::net::TcpListener::bind(&addr)
-            .await
-            .map_err(|e| Error::IO {
-                source: Box::new(e),
-                location: snafu::location!(),
-            })?;
 
-        axum::serve(listener, self.router())
-            .await
-            .map_err(|e| Error::IO {
+        let listener = tokio::net::TcpListener::bind(&addr).await.map_err(|e| {
+            log::error!("RestAdapter::start() failed to bind to {}: {}", addr, e);
+            Error::IO {
                 source: Box::new(e),
                 location: snafu::location!(),
-            })?;
+            }
+        })?;
+
+        let (shutdown_tx, mut shutdown_rx) = watch::channel(false);
+        let router = self.router();
+
+        tokio::spawn(async move {
+            let result = axum::serve(listener, router)
+                .with_graceful_shutdown(async move {
+                    let _ = shutdown_rx.changed().await;
+                })
+                .await;
 
-        Ok(())
+            if let Err(e) = result {
+                log::error!("RestAdapter: server error: {}", e);
+            }
+        });
+
+        Ok(RestAdapterHandle { shutdown_tx })
+    }
+}
+
+/// Handle for controlling a running REST adapter server.
+///
+/// Use this handle to gracefully shut down the server when it's no longer needed.
+pub struct RestAdapterHandle {
+    shutdown_tx: watch::Sender<bool>,
+}
+
+impl RestAdapterHandle {
+    /// Gracefully shut down the server.
+    ///
+    /// This signals the server to stop accepting new connections and wait for
+    /// existing connections to complete.
```

Comment:
This method doesn't actually block until the shutdown is complete though right?  I feel like we should document that fact (or ideally find some way to block?)